source/agora/consensus/protocol/Nominator.d

```diff
@@ -859,16 +859,20 @@ extern(D):
             const block = makeNewBlock(prev_block, data.tx_set, data.timestamp, data.enrolls,
                 random_seed, data.missing_validators);
 
-            // If we did not sign yet then add signature
+            // If we did not sign yet then add signature and gossip to other nodes
             if (this.schnorr_pair.V !in this.slot_sigs[height])
             {
                 log.trace("ADD BLOCK SIG at height {} for this node {}", height, this.node_public_key);
                 this.slot_sigs[height][this.schnorr_pair.V] = this.createBlockSignature(block);
             }
             const signed_block = this.updateMultiSignature(block);
-            assert(signed_block != Block.init,
-                format!"INVALID Block signature. Can not externalize this block at height %s on node %s"
-                    (height, this.node_public_key));
+            if (signed_block == Block.init)
+            {
+                log.warn("Not ready to externalize this block at height {} on node {}", height, this.node_public_key);
```

Comment:
Do we need to show our own node's public key here ?

Reply:
It has been useful when network unit tests failed to quickly identify which node it is.